HW-2 Model

juan-rybczinski opened the pull request:
- 1주차 과제 밀린 내용들 함께 적용했습니다.
- 수업 시간에 들은 캐싱 전략을 적용해 보려고 했는데 어떻게 해야할 지 잘 모르겠네요 ;

juan-rybczinski committed:
HW-1 Change the location of companion object block to the end of App class
- https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/882#discussion_r518679812

juan-rybczinski committed:
HW-1 Apply Lifecycle-aware coroutine scopes
- https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/882#discussion_r518684191
- https://developer.android.com/topic/libraries/architecture/coroutines#lifecyclescope

juan-rybczinski committed:
HW-1 Change to access view objects in onViewCreated lifecycle
- https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/882#discussion_r518684874
- https://stackoverflow.com/questions/25119090/difference-between-oncreateview-and-onviewcreated-in-fragment

juan-rybczinski committed:
HW-1 Encapsulate movies in MovieAdpater
- https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/882#discussion_r518685656

juan-rybczinski committed:
HW-1 Apply layout_margin
- https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/882#discussion_r518686245

juan-rybczinski committed:
HW-1 Apply layout_marginHorizontal
- https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/882#discussion_r518686452

juan-rybczinski committed:
HW-1 Make Movie class nested in MovieResponse class

juan-rybczinski committed:
HW-1 Apply ViewBinding
- https://developer.android.com/topic/libraries/view-binding#fragments

juan-rybczinski committed:
HW-2 Rearrange project structure
- Add domain package and put Movie class inside it
- Rename MovieResponse to NetworkMovie
- Add MovieDataTransferObjects file and put NetworkMovie class inside it

juan-rybczinski committed:
HW-2 Add an offline cache
- Add the Room dependency
- Add database object
- Add MovieDao
- https://codelabs.developers.google.com/codelabs/kotlin-android-training-repository/index.html?index=..%2F..android-kotlin-fundamentals#4

juan-rybczinski committed:
HW-2 Implement RemoteDataSource to retieve data from network

juan-rybczinski committed:
HW-2 Retrieve data by Repository

juan-rybczinski committed:
Add LocalDataSource to Repository

juan-rybczinski committed:
HW-2 Apply interface inheritance on MovieRepository

hunihun commented on class14/juan-rybczinski/GoogryAndroidArchitectureStudy/app/src/main/java/com/example/googryandroidarchitecturestudy/ui/fragment/MovieFragment.kt:
```diff
@@ -2,76 +2,68 @@ package com.example.googryandroidarchitecturestudy.ui.fragment
-                    val movieResponse = MovieApi.movieService.searchMovies(search_text.text.toString())
-                    movieAdapter.setMovies(movieResponse.items ?: emptyList())
+                    movieAdapter.setMovies(
+                        movieRepository.searchMovies(
```
api 통신중 fail 처리를 어떻게 하시는건가용 ...!? 찾아봐도 안보이네요!!

juan-rybczinski replied:
Coroutine 사용하면 `try catch`로 통신 중 fail 처리 가능한 것 같아요.
https://codelabs.developers.google.com/codelabs/kotlin-android-training-internet-data/index.html?index=..%2F..android-kotlin-fundamentals#5

hunihun commented on class14/juan-rybczinski/GoogryAndroidArchitectureStudy/app/src/main/java/com/example/googryandroidarchitecturestudy/ui/fragment/MovieFragment.kt:
```diff
@@ -2,76 +2,68 @@ package com.example.googryandroidarchitecturestudy.ui.fragment
-        job.cancel()
-
+    override fun onDestroyView() {
+        super.onDestroyView()
```
onDestroy 는 작업을 다 한 후 마지막에 super.onDestroy()를 사용해야 한다고합니다!
https://qastack.kr/programming/18821481/what-is-the-correct-order-of-calling-superclass-methods-in-onpause-onstop-and-o

juan-rybczinski replied:
반영했습니다!

baf1b326e

hunihun reviewed (commented):
Room database를 사용을 안해봐서 자세히 리뷰 못한점 죄송합니다 ㅠㅠ 공부해볼께요!

juan-rybczinski committed:
HW-2 Change to call super function after overrided logic in onDestroyView
- Fix https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/894#discussion_r520632227

hunihun commented:
고생하셨씁니다!!

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
